doc/contributing/adding-v8-fast-api.md

```diff
@@ -0,0 +1,120 @@
+# Adding V8 Fast API
+
+Node.js uses [V8](https://github.com/v8/v8) as the JavaScript engine.
+In order to provide fast paths for functions that are called quite often,
+V8 proposes the usage of `fast api calls` that does not use any of the V8
+internals, but uses internal C functions.
```

Comment:
`internal C functions` is rather ambiguous. Also, `V8 internals` usually refers to the non-public V8 API, which we try not to use in any case. Lastly, if I remember correctly, V8 API calls are allowed to some degree. Not performing any allocations on the V8 heap is what's critical. (Unless this changed since I last worked on fast API calls.)

Reply:
I see. Thanks! Can you give an example of V8 API call, without any allocation on the V8 heap?

Reply:
I might be wrong, but I think @MayaLekova implemented a fast API that relies on `GetAlignedPointerFromInternalField` in 6cfba9f7f6a54c539f10966e87dff70ba80ab6c6, for example. @devsnek might know more about the exact conditions for which APIs are allowed.

Reply:
You are right. You can access pre-existing/pre-allocated data using the fast apis, inferred from the documentation (at least I assumed). Is there any specific wording, you'd like me to change?

Reply:
I've updated the documentation.

Reply:
> [...] Not performing any allocations on the V8 heap is what's critical.

Indeed that's still the case, and until we have the current GC architecture, it will stay like this. Possible future relaxation of this rule might come with conservative stack scanning. Please note that this also leads to the limitation for not calling back into JS code (I think this was discussed in this thread too).

Some brief background on the support for various types in V8 could be found in this [external doc](https://docs.google.com/document/d/1nK6oW11arlRb7AA76lJqrBIygqjgdc92aXUPYecc9dU/edit?usp=sharing).